### PR TITLE
feat(plugin): new Row Based Editing

### PR DESCRIPTION
**Pull request by ghiscoding**

### CODE REVIEW

File: src/app/examples/grid-base-row-editing.component.ts [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+  constructor(private translate: TranslateService) {
+    // always start with English for Cypress E2E tests to be consistent
+    const defaultLang = 'en';
+    this.translate.use(defaultLang);
+    this.selectedLanguage = defaultLang;
+  }
+
+  ngOnInit() {
+    // mock some data (different in each dataset)
```
Comment (zewa666) [inline src/app/examples/grid-base-row-editing.component.ts:42]: this comment is a copy leftover from sample 1

Comment (ghiscoding) [reply src/app/examples/grid-base-row-editing.component.ts]: ah yeah didn't notice, I went rather quickly before going at the gym. thanks for the review

**Comment by ghiscoding** [comment]
@zewa666 I've pushed a new release but it's getting late and I just realized that I forgot to add the documentation changes, if you could do a PR, that would be great. If not, I'll do it on Monday. Cheers

**Comment by ghiscoding** [comment]
@zewa666 I don't think that the translation are working as expected, typically when there's a translation key provided well it should use it, otherwise use prop without the key suffix but only as a last resource. It doesn't seem to be translating at all even in Slickgrid-Universal either, however the header menu and such are translated correctly

image

**Comment by zewa666** [comment]
hmmm must have been an issue while refactoring. I'll check that out tomorrow. thx for the Feedback

**Comment by ghiscoding** [comment]
@zewa666 
dohhh I just realized that you don't translate the first 2 default buttons shown in the screen but the next 2 (after editing) are being translated. I think that is because of how you've setup the example which was a big confusing. 

Nonetheless, I see some perf improvement that I can look into (not just in the RowBase plugin but also in the ExtensionService), there's couple of calls to translate/invalidate more than once but it really should be 1 invalidate and it should be **after** the translation is done... So I'll take care of it, nothing to do on your side apart from a possible PR review once it's open

msedge_3EU7sRwqWy

**Comment by zewa666** [comment]
oh crap yeah now I remember. I didnt want to update all of them to let the user see how things gradually fallback from key to title to default. sry for the confusion. yes plz ping me on the PR for a review

**Pull request merged**